src/core/TimeTypes.re

```diff
@@ -12,27 +12,43 @@ type timeUnit = [
 
 type timeVector = {
   zero: MomentRe.Moment.t,
-  unit: timeUnit,
+  length: MomentRe.Moment.t,
```

Comment:
What's the purpose of "length" here? I was assuming that the notion of length would be captured by the X values of the distribution.

Sorry this wasn't very explained.

Reply:
So in this way how to get the end date?

Reply:
Suppose we have x: [10...1000] and the zero date as this day, so how to get the end date?

Reply:
It's a bit tricky. The File TimeLimitedDomainCdf (which may have been deleted here) shows how do do it. Basically, we'd want to get the first and last values of the X Pdf, and multiply them by the time axis, compared to the zero point.

So, the pdf value of x=-24, with a time vector of {zero: Jan 1 2000, unit: `day}, would correspond to 24 days before Jan 1 2000. We'd need to do this calculation for the first and last x-values of the pdf.

For now, maybe use have the start and end dates passed in from a higher component.